src/xenia/hid/winkey/winkey_input_driver.cc

```diff
@@ -10,17 +10,121 @@
+DEFINE_string(keybind_left_thumb_left,   "_'A'",    "XINPUT_GAMEPAD_LEFT_THUMB_LEFT"  , "KeyBind")
+DEFINE_string(keybind_left_thumb_right,  "_'D'",    "XINPUT_GAMEPAD_LEFT_THUMB_RIGHT" , "KeyBind")
+DEFINE_string(keybind_left_thumb_down,   "_'S'",    "XINPUT_GAMEPAD_LEFT_THUMB_DOWN"  , "KeyBind")
+DEFINE_string(keybind_left_thumb_up,     "_'W'",    "XINPUT_GAMEPAD_LEFT_THUMB_UP"    , "KeyBind")
+DEFINE_string(keybind_left_thumb,        "'F'",     "XINPUT_GAMEPAD_LEFT_THUMB"       , "KeyBind")
+DEFINE_string(keybind_right_thumb_up,    "0x26",    "XINPUT_GAMEPAD_RIGHT_THUMB_UP"   , "KeyBind")
+DEFINE_string(keybind_right_thumb_down,  "0x28",    "XINPUT_GAMEPAD_RIGHT_THUMB_DOWN" , "KeyBind")
+DEFINE_string(keybind_right_thumb_right, "0x27",    "XINPUT_GAMEPAD_RIGHT_THUMB_RIGHT", "KeyBind")
+DEFINE_string(keybind_right_thumb_left,  "0x25",    "XINPUT_GAMEPAD_RIGHT_THUMB_LEFT" , "KeyBind")
+DEFINE_string(keybind_right_thumb,       "'K'",     "XINPUT_GAMEPAD_RIGHT_THUMB"      , "KeyBind")
+DEFINE_string(keybind_x,                 "'L'",     "XINPUT_GAMEPAD_X"                , "KeyBind")
+DEFINE_string(keybind_b,                 "0xDE",    "XINPUT_GAMEPAD_B"                , "KeyBind")
+DEFINE_string(keybind_a,                 "0xBA",    "XINPUT_GAMEPAD_A"                , "KeyBind")
+DEFINE_string(keybind_y,                 "'P'",     "XINPUT_GAMEPAD_Y"                , "KeyBind")
+DEFINE_string(keybind_left_trigger,      "'Q' 'I'", "XINPUT_GAMEPAD_LEFT_TRIGGER"     , "KeyBind")
+DEFINE_string(keybind_right_trigger,     "'E' 'O'", "XINPUT_GAMEPAD_RIGHT_TRIGGER"    , "KeyBind")
+DEFINE_string(keybind_back,              "'Z'",     "XINPUT_GAMEPAD_BACK"             , "KeyBind")
+DEFINE_string(keybind_start,             "'X'",     "XINPUT_GAMEPAD_START"            , "KeyBind")
+DEFINE_string(keybind_left_shoulder,     "'1'",     "XINPUT_GAMEPAD_LEFT_SHOULDER"    , "KeyBind")
+DEFINE_string(keybind_right_shoulder,    "'3'",     "XINPUT_GAMEPAD_RIGHT_SHOULDER"   , "KeyBind")
+
 namespace xe {
 namespace hid {
 namespace winkey {
 
+void WinKeyInputDriver::ParseKeyBindingCVar(ui::VirtualKey controllerKey, std::string keyBindCVar) {
+  const char *cstr = keyBindCVar.c_str();
+  char token[16];
+  std::size_t off = 0;
+  while (keyBindCVar[off] != 0) {
+    std::size_t end = keyBindCVar.find(' ', off);
+    if (end == std::string::npos) {
+      end = keyBindCVar.length();
+    }
+    if (end - off < 15) {
+      memcpy(token, keyBindCVar.c_str() + off, end - off);
+      token[end - off] = 0;
+
+      char *s = token;
+      KeyBinding bind;
+      if (*s == '_') {
+        bind.lowercase = true;
+        s++;
+      }
+      if (*s == '^') {
+        bind.uppercase = true;
+        s++;
+      }
+      if (s[0] == '\'' && s[1] != '\0') {
+        bind.key = (ui::VirtualKey)(uint8_t) s[1];
+      } else {
+        char *endptr;
+        unsigned long ul = strtoul(s, &endptr, 0);
```

Comment:
Don't use `strtoul`, we have helper functions for this in `base/string_util.h` (see `from_string`).